Xamarin.Forms.Platform.iOS/Renderers/NavigationRenderer.cs

```diff
@@ -622,45 +617,65 @@ void UpdateBackgroundColor()
 		{
 			var barBackgroundColor = NavPage.BarBackgroundColor;
-			// Set navigation bar background color
-			NavigationBar.BarTintColor = barBackgroundColor == Color.Default
+
+			if (!Forms.IsiOS13OrNewer)
+			{
+				// Set navigation bar background color
+				NavigationBar.BarTintColor = barBackgroundColor == Color.Default
 				? UINavigationBar.Appearance.BarTintColor
 				: barBackgroundColor.ToUIColor();
+			}
+			else
+			{
+				var navigationBarAppearance = new UINavigationBarAppearance();
+
+				if (barBackgroundColor == Color.Default)
+					navigationBarAppearance.ConfigureWithDefaultBackground();
+				else
+				{
+					navigationBarAppearance.ConfigureWithOpaqueBackground();
+					navigationBarAppearance.BackgroundColor = barBackgroundColor.ToUIColor();
+				}
+
+				NavigationBar.CompactAppearance = navigationBarAppearance;
+				NavigationBar.StandardAppearance = navigationBarAppearance;
+				NavigationBar.ScrollEdgeAppearance = navigationBarAppearance;
+			}
 		}
 
 		void UpdateBarTextColor()
 		{
 			var barTextColor = NavPage.BarTextColor;
-
 			var globalAttributes = UINavigationBar.Appearance.GetTitleTextAttributes();
+			var titleTextAttributes = new UIStringAttributes
+			{
+				ForegroundColor = barTextColor == Color.Default ? globalAttributes.TextColor : barTextColor.ToUIColor(),
+				Font = globalAttributes.Font
+			};
 
-			if (barTextColor == Color.Default)
+			if (Forms.IsiOS13OrNewer)
 			{
-				if (NavigationBar.TitleTextAttributes != null)
-				{
-					var attributes = new UIStringAttributes();
-					attributes.ForegroundColor = globalAttributes.TextColor;
-					attributes.Font = globalAttributes.Font;
-					NavigationBar.TitleTextAttributes = attributes;
-				}
+				NavigationBar.CompactAppearance.TitleTextAttributes = titleTextAttributes;
+				NavigationBar.CompactAppearance.LargeTitleTextAttributes = titleTextAttributes;
+
+				NavigationBar.StandardAppearance.TitleTextAttributes = titleTextAttributes;
+				NavigationBar.StandardAppearance.LargeTitleTextAttributes = titleTextAttributes;
+
+				NavigationBar.ScrollEdgeAppearance.TitleTextAttributes = titleTextAttributes;
+				NavigationBar.ScrollEdgeAppearance.LargeTitleTextAttributes = titleTextAttributes;
 			}
 			else
 			{
-				var titleAttributes = new UIStringAttributes();
-				titleAttributes.Font = globalAttributes.Font;
-				// TODO: the ternary if statement here will always return false because of the encapsulating if statement.
-				// What was the intention?
-				titleAttributes.ForegroundColor = barTextColor == Color.Default
-					? titleAttributes.ForegroundColor ?? UINavigationBar.Appearance.TintColor
-					: barTextColor.ToUIColor();
-				NavigationBar.TitleTextAttributes = titleAttributes;
-			}
+				if (barTextColor == Color.Default)
+				{
+					if (NavigationBar.TitleTextAttributes != null)
+						NavigationBar.TitleTextAttributes = titleTextAttributes;
+				}
+				else
+					NavigationBar.TitleTextAttributes = titleTextAttributes;
```

Comment:
Lines 669 - 675 can possibly be simplified by `NavigationBar.TitleTextAttributes = titleTextAttributes;`, but I'm not sure why the previous code had that extra null check on `NavigationBar.TitleTextAttributes`.